GH-3557: Add maxDepth, dirPredicate to FileReadMS (#8596)
spring-projects/spring-integration: src/reference/asciidoc/file.adoc

----
====

Starting with version 6.1, the `FileReadingMessageSource` exposes two new `WatchService`-related options:

* `watchMaxDepth` - an argument for the `Files.walkFileTree(Path root, Set attributes, int maxDepth, FileVisitor visitor)` API;

> * `watchDirPredicate` - a `Predicate<Path>` to test if directory in the scanned tree should be walked and registered for `WatchService` with its modification events.

Reviewer: ```suggestion
* `watchDirPredicate` - a `Predicate<Path>` to test if a directory in the scanned tree should be walked and registered with the `WatchService` and the configured watch event kinds.
```
